lib/internal/util.js

```diff
@@ -203,6 +203,42 @@ function getConstructorOf(obj) {
   return null;
 }
 
+function getIdentifiableTagOf(obj) {
+  const original = obj;
+  let constructor = '';
+  let tag = '';
+
+  while (obj) {
+    if (constructor === '') {
+      var desc = Object.getOwnPropertyDescriptor(obj, 'constructor');
+      if (desc !== undefined &&
+         typeof desc.value === 'function' &&
+         desc.value.name !== '')
+        constructor = desc.value.name;
+    }
+
+    if (tag === '') {
+      desc = Object.getOwnPropertyDescriptor(obj, Symbol.toStringTag);
+      if (desc !== undefined) {
+        if (desc.value !== undefined)
```

Comment:
`typeof desc.value` should be checked for `'string'`. Non-strings should be ignored.